pkg/conference/matrix_worker.go

```diff
@@ -0,0 +1,46 @@
+package conference
+
+import (
+	"time"
+
+	"github.com/matrix-org/waterfall/pkg/common"
+	"github.com/matrix-org/waterfall/pkg/signaling"
+	"github.com/sirupsen/logrus"
+	"maunium.net/go/mautrix/id"
+)
+
+type matrixWorker struct {
+	worker   *common.Worker[signaling.MatrixMessage]
+	deviceID id.DeviceID
+}
+
+func newMatrixWorker(handler signaling.MatrixSignaler) *matrixWorker {
+	workerConfig := common.WorkerConfig[signaling.MatrixMessage]{
+		ChannelSize: 128,
+		Timeout:     time.Hour,
+		OnTimeout:   func() {},
+		OnTask:      func(msg signaling.MatrixMessage) { handler.SendMessage(msg) },
+	}
+
+	matrixWorker := &matrixWorker{
+		worker:   common.StartWorker(workerConfig),
+		deviceID: handler.DeviceID(),
+	}
+
+	return matrixWorker
+}
+
+func (w *matrixWorker) stop() {
+	w.worker.Stop()
+}
+
+func (w *matrixWorker) sendSignalingMessage(recipient signaling.MatrixRecipient, content interface{}) {
+	msg := signaling.MatrixMessage{
+		Recipient: recipient,
+		Message:   content,
+	}
+
+	if err := w.worker.Send(msg); err != nil {
+		logrus.Panic("Really bad, dropping matrix message since the matrix queue is full! Home server down?")
```

Comment:
Is it bad enough to panic the whole SFU though?

Reply:
Ah, you're right, I assumed `Panic` would just print a critical error or something (`Fatal` is documented to exit the process, so I thought that `Panic` would be `Critical` or something, but then checked the source code an found out that it panics). Will fix it right away!

Reply:
Fixed.